### PR TITLE
Added support for mocked commands in tests and test for govcms-db-update script.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
WIP/POC DO NOT MERGE

- Added binary mocker for BATs (includes some adjustments)
- Added BATs assertion helpers.
- Added tests for `govcms-db-update` script.

scaffold-tooling____www_scaffold-tooling__-_____tests_bats_deploy_update_bats__scaffold-tooling_

### CODE REVIEW

File: .circleci/shellcheck.sh [hunk 1]
```diff
@@ -12,8 +12,8 @@ done < <(
     .circleci/bats.sh \
     .circleci/phpcs.sh \
     .circleci/shellcheck.sh \
-    ! -name "README.*" \
     -type f \
+    \( ! -name "README.*" -and ! -name "_bats-mock.bash" \) \
```
Comment (AlexSkrypnyk) [inline .circleci/shellcheck.sh:16]: do not lint `_bats-mock.bash` as it is a 3rd party library

File: tests/bats/_bats-mock.bash [hunk 2]
```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env bash
```
Comment (AlexSkrypnyk) [inline tests/bats/_bats-mock.bash:1]: this is a 3rd party library.

Unfortunately, it has to be committed directly.

File: tests/bats/_helpers.bash [hunk 3]
```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env bash
```
Comment (AlexSkrypnyk) [inline tests/bats/_helpers.bash:1]: common helpers. these were tested in DrevOps project (have their own tests). 
https://github.com/drevops/drevops/blob/8.x/tests/bats/helpers.bats

File: tests/bats/deploy/_helpers_deploy.bash [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+load ../_helpers
+load ../_bats-mock
+
+setup() {
```
Comment (AlexSkrypnyk) [inline tests/bats/deploy/_helpers_deploy.bash:6]: this is the magic that makes mocked binaries work (not as easy as it should be out of the box)

File: tests/bats/deploy/_helpers_deploy.bash [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  # Prepare directory with mock binaries, get it's path, and export it so that
+  # bats-mock could use it internally.
+  BATS_MOCK_TMPDIR="$(mock_prepare_tmp)"
+  export "BATS_MOCK_TMPDIR"
+  # Set path to temp binaries directory as the first location in PATH to
+  # lookup in mock directories first.
+  PATH="${BATS_MOCK_TMPDIR}:$PATH"
```
Comment (AlexSkrypnyk) [inline tests/bats/deploy/_helpers_deploy.bash:13]: this change lives only for the duration of the test and will be reset after.

**Comment by AlexSkrypnyk** [comment]
@steveworley could you please advise if these tests is something that would be useful. If so - I can add test coverage for all of the deployment scripts.

**Comment by simesy** [comment]
👍

**Comment by srowlands** [comment]
Love this Alex, very keen on more of this. +1 from me too

**Pull request merged**